zulipterminal/ui_tools/views.py

```diff
@@ -950,14 +950,29 @@ def keypress(self, size: urwid_Size, key: str) -> str:
 
 class StreamInfoView(PopUpView):
     def __init__(self, controller: Any, color: str,
-                 desc: str, title: str) -> None:
+                 desc: str, stream_id: int, title: str) -> None:
         # Add 4 (for 2 Unicode characters on either side) to the popup title
         # length to make sure that the title gets displayed even when the
         # content is shorter than the title length (+4 Unicode characters).
-        width = max(len(desc) + 2, len(title) + 4)
-        stream_info_content = [urwid.Text(desc, align='center')]
-        super().__init__(controller, stream_info_content, 'STREAM_DESC', width,
-                         title)
+        # 20 is the length required to show stream settings labels.
+        width = max(len(desc) + 2, len(title) + 4, 20)
+        self.controller = controller
+        self.stream_id = stream_id
+        muted_setting = urwid.CheckBox(label="Mute stream",
+                                       state=controller.model.is_muted_stream(
+                                           stream_id),
+                                       checked_symbol='✓')
+        urwid.connect_signal(muted_setting, 'change', self.toggle_mute_status)
+
+        # Include blank line between description and stream settings
+        self.stream_info_content = [urwid.Text(desc, align='center'),
+                                    urwid.Text('', align='center'),
+                                    muted_setting]
+        super().__init__(controller, self.stream_info_content, 'STREAM_DESC',
+                         width, title)
+
+    def toggle_mute_status(self, button: Any, new_state: bool) -> None:
```

Comment:
`new_state` is unused; did this code come from somewhere else? Should we name it accordingly or elide it, if it's unused?

Reply:
It's urwids default function signature for callback functions